modules/collections/client/controllers/collections.controllers.js

```diff
@@ -159,145 +164,210 @@ collectionModules.controller('CollectionEditCtrl',
             $state.reload();
           })
-          .catch(function(/* res */) {
+          .catch(function() {
+            // User cancels publish, or error is thrown.
             AlertService.error('"'+ $scope.collection.displayName +'" was not unpublished.', 4000, true);
-            $state.reload();
           });
       };
 
-      $scope.save = function(isValid) {
-        if (!isValid) {
-          $scope.$broadcast('show-errors-check-validity', 'collectionForm');
-          AlertService.error('Error: The project could not be saved. Ensure project has a type, unique name, and description.', 4000, true);
-          return false;
+      // Validate, and save the collection.
+      $scope.save = function() {
+        if ($scope.collection.isPublished) {
+          $scope.checkValidity(true);
+        } else {
+          $scope.checkValidity(false);
+        }
+        if ($scope.validationFlags.detailsTabInvalid || $scope.validationFlags.documentsTabInvalid) {
+          return;
         }
-        // Update parent and status
+        // Update parent and status.
         $scope.collection.status = 'Issued';
         switch($scope.collection.type) {
         case 'Permit Amendment':
           $scope.collection.parentType = 'Authorizations';
           $scope.collection.status = 'Amended';
           break;
-
         case 'Permit':
           $scope.collection.parentType = 'Authorizations';
           break;
-
         case 'Inspection Report':
         case 'Order':
           $scope.collection.parentType = 'Compliance and Enforcement';
           break;
-
         case 'Annual Report':
         case 'Management Plan':
         case 'Dam Safety Inspection':
         case 'Letter of Assurance':
           $scope.collection.parentType = 'Other';
           break;
         }
-        CollectionModel.save($scope.collection)
-          .then(function (/* model */) {
-            AlertService.success('"' + $scope.collection.displayName + '" was saved successfully.', 4000, true);
-            $state.reload();
-          }, function () {
-            AlertService.error('"' + $scope.collection.displayName + '" was unable to save successfully.', 4000, true);
-            $state.reload();
-          })
-          .catch(function(/* err */) {
-            // swallow rejected promise error
-          });
+        if ($scope.collection.addedBy) {
+          // Editing an existing collection.
+          CollectionModel.save($scope.collection)
+            .then(function(){
+              if ($scope.validationFlags.docsChanged) {
+                $scope.submitDocs(true);
+              } else {
+                AlertService.success('"' + $scope.collection.displayName + '" was saved successfully.', 4000, true)
+                $scope.goToEdit();
+              }
+            })
+            .catch(function() {
+              AlertService.error('"' + $scope.collection.displayName + '" was unable to save successfully.', 4000, true);
```

Comment:
I know this was here before, but "was unable to save successfully" reads a bit funny to me.
Do our UX folk have any thoughts?

Reply:
Pretty sure that IS from our UX people lol